pika/adapters/base_connection.py

```diff
@@ -134,9 +134,23 @@ def _adapter_disconnect(self):
         if self.socket:
             self.socket.close()
         self.socket = None
-        self._check_state_on_disconnect()
-        self._handle_ioloop_stop()
-        self._init_connection_state()
+
+        try:
+            self._check_state_on_disconnect()
+        except exceptions.AMQPConnectionError as e:
```

Comment:
> The title of the PR refers to ProbableAuthenticationError, but the exception being trapped here is more general.

Yes, that is true. I observed the problem only with the ```ProbableAuthenticationError```. IMO, however, the ```on_open_error_callback``` should be called for any connection errors in cases where the connection has not been fully opened.